estuary/ui
# Preserve `sourceCapture` settings of an existing materialization

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/874

## Changes

### 874

The following features are included in this PR:

* Use the appropriate entity workflow hook to determine whether the user is in an edit workflow inside the resource config store hydrator.

* Prevent the `sourceCapture` settings of an existing materialization from being overridden when it is used to materialize a single capture.

## Tests

### Manually tested

Approaches to testing are as follows:

* Select a single row of the captures table on the _Captures_ page, click the _Materialize_ CTA, enter the materialization create workflow, and observe that the `sourceCapture` property of the drafted specification is set to the selected capture name.

* Select multiple rows of the captures table on the _Captures_ page, click the _Materialize_ CTA, enter the materialization create workflow, and observe that a `sourceCapture` property does not exist in the drafted specification.

* Select a single row of the captures table on the _Captures_ page, click the _Materialize_ CTA, select a connector for which a materialization exists, select an existing materialization on the existing entity check page, enter the materialization edit workflow, and observe that the `sourceCapture` settings remain unchanged.

* Select multiple rows of the captures table on the _Captures_ page click the _Materialize_ CTA, select a connector for which a materialization exists, select an existing materialization on the existing entity check page, enter the materialization edit workflow, and observe that the `sourceCapture` settings remain unchanged.

### Automated tests

-   None

## Screenshots

N/A
